datacube/metadata/_eo3converter.py

```diff
@@ -0,0 +1,356 @@
+    expected to have the same set of bands. If product contains bands with different resolutions, it
+    is assumed that the same set of bands share common resolution across all items in the
+    collection.
+
+    :param items:
+       Lazy sequence of :class:`~pystac.item.Item` objects
+
+    :param cfg:
+       Supply metadata missing from STAC, configure aliases, control warnings
+
+    :param product_cache:
+       Input/Output parameter, contains mapping from collection name to deduced product definition,
+       i.e. :py:class:`datacube.model.Product` object.
+
+    .. rubric: Sample Configuration
+
+    .. code-block:: yaml
+
+       sentinel-2-l2a:  # < name of the collection, i.e. `.collection_id`
+         assets:
+           "*":  # Band named "*" contains band info for "most" bands
+             data_type: uint16
+             nodata: 0
+             unit: "1"
+           SCL:  # Those bands that are different than "most"
+             data_type: uint8
+             nodata: 0
+             unit: "1"
+         aliases:  #< unique alias -> canonical map
+           rededge: B05
+           rededge1: B05
+           rededge2: B06
+           rededge3: B07
+         uuid:          # Rules for constructing UUID for Datasets
+           mode: auto   # auto|random|native(expect .id to contain valid UUID string)
+           extras:      # List of extra keys from properties to include (mode=auto)
+           - "s2:generation_time"
+
+         warnings: ignore  # ignore|all  (default all)
+
+       some-other-collection:
+         assets:
+         #...
+
+       "*": # Applies to all collections if not defined on a collection
+         warnings: ignore
+
+    """
+    products: dict[str, Product] = {} if product_cache is None else product_cache
+    for item in items:
+        collection_id = _collection_id(item)
+        product = products.get(collection_id)
+
+        # Have not seen this collection yet, figure it out
+        if product is None:
+            product = infer_dc_product(item, cfg)
+            products[collection_id] = product
+
+        yield _item_to_ds(item, product, cfg)
+
+
+@infer_dc_product.register(pystac.collection.Collection)
+def infer_dc_product_from_collection(
+    collection: pystac.collection.Collection, cfg: ConversionConfig | None = None
+) -> Product:
+    """
+    Construct Datacube Product definition from STAC Collection.
+
+    :param collection: STAC Collection
+    :param cfg: Configuration dictionary
+    """
+    # pylint: disable=protected-access
+    if cfg is None:
+        cfg = {}
+    product = infer_dc_product(mk_sample_item(collection), cfg)
+
+    # unless configured to ignore projection info assume that it will be present
+    ignore_proj = cfg.get(product.name, {}).get("ignore_proj", False)
+    if not ignore_proj:
+        product._md = dataclasses.replace(product._md, has_proj=True)  # type: ignore
```

Comment:
Can the ignore get a `ignore[attr-defined]` or whatever problem  this is ignoring?

Reply:
Not sure - there are a few bits of mypy weirdness going on that I don't fully understand yet.

Reply:
@SpacemanPaul see also this https://github.com/opendatacube/odc-stac/pull/218/files#diff-5643a2305b1d8d68e811bc7844fecdce8077079bfc1675a66deb16fb2334fcba

Reply:
This code is shoving internal stuff into the product model.  Should probably integrate better seeing as we are in the same repo now.

@Kirill888 I've added TODOs to update when loader 0.6.0 is released.